snuba/state/__init__.py

```diff
@@ -313,7 +311,28 @@ def record_query(query_metadata: Mapping[str, Any]) -> None:
         ).execute()
 
         if kfk is None:
-            kfk = Producer(build_default_kafka_producer_configuration())
+            kfk = Producer(
+                # the querylog payloads can get really large so we allow larger messages
+                # (double the default)
+                # The performance is not business critical and therefore we accept the tradeoffs
+                # in more bandwidth for more observability/debugability
+                # for this to be meaningful, the following setting has to be matched on the broker:
+                # max.message.bytes=2097176
+                build_kafka_producer_configuration(
+                    topic=None,
+                    override_params={
+                        # at time of writing (2022-05-09) lz4 was chosen because it
+                        # compresses quickly. If more compression is needed at the cost of
+                        # performance, zstd can be used instead. Recording the query
+                        # is part of the API request, therefore speed is important
+                        # perf-testing: https://indico.fnal.gov/event/16264/contributions/36466/attachments/22610/28037/Zstd__LZ4.pdf
+                        # by default a topic is configured to use whatever compression method the producer used
+                        # https://docs.confluent.io/platform/current/installation/configuration/topic-configs.html#topicconfigs_compression.type
+                        "compression.type": "lz4",
+                        "max.request.size": 2097176,
```

Comment:
Where does `max.request.size` come from? I didn't see it in the list of librdkafka configurations

Reply:
Our other producers use `message.max.bytes`

Reply:
https://docs.confluent.io/platform/current/installation/configuration/producer-configs.html#producerconfigs_max.request.size